src/components/KSelect/KSelect.vue

```diff
@@ -2,278 +2,195 @@
   <div
     class="k-select"
     :class="[$attrs.class]"
-    :style="widthStyle"
   >
-    <KLabel
-      v-if="label && !overlayLabel"
-      v-bind="labelAttributes"
-      data-testid="k-select-label"
-      :for="selectId"
-      :required="isRequired"
-    >
-      {{ strippedLabel }}
-
-      <template
-        v-if="hasLabelTooltip"
-        #tooltip
-      >
-        <slot name="label-tooltip" />
-      </template>
-    </KLabel>
-    <div
-      :id="selectId"
-      data-testid="k-select-selected-item"
-    >
-      <div
-        v-if="selectedItem && appearance === 'dropdown'"
-        class="k-select-item-selection"
-        :class="{ 'overlay-label-item-selection': overlayLabel }"
-      >
-        <slot
-          :item="selectedItem"
-          name="selected-item-template"
-        >
-          <div
-            class="k-select-selected-item-label"
-          >
-            {{ selectedItem.label }}
-          </div>
-        </slot>
-        <button
-          class="clear-selection-icon"
-          @click="clearSelection"
-          @keyup.enter="clearSelection"
-        >
-          <KIcon
-            color="currentColor"
-            icon="clear"
-          />
-        </button>
-      </div>
-      <KToggle v-slot="{ toggle, isToggled }">
-        <KPop
-          ref="popper"
-          v-bind="boundKPopAttributes"
-          :on-popover-click="() => {
-            toggle()
-            return isToggled.value
-          }"
-          :position-fixed="positionFixed"
-          :target="`[id='${selectInputId}']`"
-          :test-mode="!!testMode || undefined"
-          @closed="() => {
-            if (selectedItem && appearance === 'select') {
-              filterStr = selectedItem.label
-            }
-            if (isToggled.value) {
-              toggle()
-            }
-          }"
-          @opened="() => {
-            filterStr = ''
+    <KToggle v-slot="{ toggle, isToggled }">
+      <KPop
+        ref="popperElement"
+        v-bind="boundKPopAttributes"
+        :on-popover-click="() => {
+          toggle()
+          return isToggled.value
+        }"
+        :position-fixed="positionFixed"
+        target=".select-wrapper"
+        @closed="() => {
+          if (selectedItem) {
+            filterQuery = selectedItem.label
+          }
+          if (isToggled.value) {
```

Comment:
suggestion: would it be cleaner to refactor these in-template function blocks `@closed` `on-popover-click` etc. to functions down in the script block? You should be able to pass around the diff callbacks/vars as needed; just wondering if it would be cleaner.